generators/spring-boot/mixin.cjs

```diff
@@ -17,21 +17,29 @@
  * limitations under the License.
  */
 const { defaults } = require('lodash');
+
 const { requiredConfig, defaultConfig } = require('./config.cjs');
 const { options } = require('./options.cjs');
-const { SPRING_BOOT_VERSION, SPRING_BOOT_PARENT_BOM } = require('./constants.cjs');
+const { SPRING_BOOT_VERSION, SPRING_BOOT_PARENT_BOM, SPRING_BOOT_PARENT_BOM_DEFAULT_VALUE } = require('./constants.cjs');
 
 const { GENERATOR_JAVA } = require('../generator-list');
 
 module.exports.dependencyChain = [GENERATOR_JAVA];
 
 module.exports.mixin = parent =>
   class extends parent {
+    /**
+     * Load spring-boot options constants.
+     */
+    loadSpringBootOptionsConstants(into = this) {
+      into.SPRING_BOOT_PARENT_BOM_DEFAULT_VALUE = SPRING_BOOT_PARENT_BOM_DEFAULT_VALUE;
+    }
+
     /**
      * Register and parse spring-boot options.
      */
-    registerSpringBootOptions() {
-      this.jhipsterOptions(options);
+    getSpringBootOptions() {
+      return options;
```

Comment:
They cannot mix with applications and I hope mixins will not be required for entities.
What you consider a platform scope?
From my point of view, it's specific to a technology, and are hierarchical instead of polymorphic.

By changing getSpringBootOptions to getSpringBootApplicationOptions, it would make sense to change the entire mixin api to *Application*.

Reply:
Monitoring and service registry? I think that also docker-compose, kubenretes, openshift, ... Any platform configuration could be part of the platformconfig

Reply:
We should create platform generator with those configs.

Reply:
yes, but in addition I was thinking about populating a dedicated `scope` object for each kind of options instead of injecting them in 'this'.